torchgeo/trainers/autoregression.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for autoregression."""
+
+from typing import Any
+
+import torch.nn as nn
+from torch import Tensor
+from torchmetrics import MetricCollection
+from torchmetrics.regression import MeanAbsoluteError, MeanSquaredError
+
+from torchgeo.models import Seq2Seq
+
+from .base import BaseTask
+
+
+class AutoregressionTask(BaseTask):
+    """Autoregression."""
+
+    def __init__(
+        self,
+        model: str = 'seq2seq',
+        input_size: int = 1,
+        input_size_decoder: int = 1,
+        output_size: int = 1,
+        target_indices: list[int] | None = None,
+        loss: str = 'mse',
+        lr: float = 1e-3,
+        patience: int = 10,
+        **kwargs: dict[str, Any],
+    ) -> None:
+        """Initialize a new AutoregressionTask instance.
+
+        Args:
+            model: Name of the model to use, currently supports 'seq2seq'.
+                Defaults to 'seq2seq'.
+            input_size: The number of features in the input. Defaults to 1.
+            input_size_decoder: The number of features in the decoder input.
+                Defaults to 1.
+            output_size: The number of features output by the model. Defaults to 1.
+            target_indices: The indices of the target(s) in the dataset. If None, uses all features. Defaults to None.
+            loss: One of 'mse' or 'mae'. Defaults to 'mse'.
+            lr: Learning rate for optimizer. Defaults to 1e-3.
+            patience: Patience for learning rate scheduler. Defaults to 10.
+            **kwargs: Additional keyword arguments passed to the model.
+        """
+        self.kwargs: dict[str, Any] = kwargs
+        super().__init__()
+
+    def configure_models(self) -> None:
+        """Initialize the model."""
+        model: str = self.hparams['model']
+        input_size = self.hparams['input_size']
+        input_size_decoder = self.hparams['input_size_decoder']
+
+        if model == 'seq2seq':
+            self.model = Seq2Seq(
+                input_size_encoder=input_size,
+                input_size_decoder=input_size_decoder,
+                **self.kwargs,
+            )
+        else:
+            raise ValueError(
+                f"Model type '{model}' is not valid. "
+                "Currently, only supports 'seq2seq'."
+            )
+
+    def configure_losses(self) -> None:
+        """Initialize the loss criterion.
+
+        Raises:
+            ValueError: If *loss* is invalid.
+        """
+        loss: str = self.hparams['loss']
+        if loss == 'mse':
```

Comment:
match-case statement, no need to check invalid options if you don't want to, hard to get coverage